### ISSUE TITLE
Automatically Assign Uploaded File To Client

### ISSUE THREAD

**Issue by lardav75** [projectsend/projectsend#1028]
I just updated to the latest version and no files are being auto assigned to the client who is uploading the files now. In the past, any file that was uploaded by a client was auto assigned to them. Is this functionality still doable?

**Comment by ignacionelson** [comment]
Hi! There's no need to create a relation record since the current user is the owner of the file

`public function addToDatabase()
{
    $this->uploader = CURRENT_USER_USERNAME;
    $this->uploader_id = CURRENT_USER_ID;
    $this->uploader_type = CURRENT_USER_TYPE;
    $this->hidden = 0;
     ...
`

Each user can see their own files plus the ones in the relations table.
Is this not working for your clients? Or regular users?

**Comment by lardav75** [comment]
Hi Ignacio,

The problem we were seeing is the client's files are not automatically assigned to them, so an admin user can see all of the files the user uploaded. The Assigned column would show the red No if a client uploaded files when the /manage-files.php page was viewed by an admin user.

I wrote some code recently that auto assigns the file to the user so the admin user will see all files associated with the user when they visit the /clients.php page and click the blue Files button. Before adding this code, any user that added files would not show them as belonging to them as they did not have an assignment.

To address this issue, I added the following code to /includes/Classes/Files.php:

— added lines 766 through 772: insert client id into files_relations table for auto assignment upon upload

            $statement = '';
            $statement = $this->dbh->prepare("INSERT INTO " . TABLE_FILES_RELATIONS . " (file_id, client_id, hidden)"
                                                ."VALUES (:file_id, :uploader, :hidden)");
            $statement->bindParam(':file_id', $this->id, PDO::PARAM_INT);
            $statement->bindParam(':uploader', $this->uploader_id);
            $statement->bindParam(':hidden',$this->hidden, PDO::PARAM_INT);
            $statement->execute();

**Comment by csnydermph** [comment]
Seconded, this is a problem for administrating file uploads because otherwise there is no streamlined way to see the files associated with a user when they are uploading the files:

<img width="1619" height="128" alt="Image" src="https://github.com/user-attachments/assets/7ac46b2b-8f8c-44ca-9783-7eb496e70528" />

The "Files" view is only available when the uploaded files have been manually assigned to a client.

Confirming that adding the above code still fixes the issue of files not being assigned to the uploading client automatically under version r1720.

<img width="1021" height="647" alt="Image" src="https://github.com/user-attachments/assets/d7f760c5-1cf7-4cf4-badc-5ea3d4c464b2" />

**Comment by onlinestatements** [comment]
Yes I am also having a hard time viewing clients files when login in as the system user it shows the client has no files even though the number of files shown says otherwise.

I have 1720 also and really needed this.
Hopefully future release will have this merged 

Now when I hit the view as button all I get is a blank white screen. any idea? Should I post a new issue for that?

Thank You.

**Comment by ignacionelson** [comment]
This issue is being closed as part of a bulk cleanup. ProjectSend has undergone significant changes since this was reported (including r1720 and r1994), and many older issues no longer apply to the current codebase.

If you are still experiencing this problem on the latest version, please open a new issue with updated details and we'll be happy to look into it. Thank you for your contribution!
